sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -843,6 +850,217 @@ def is_fast_reboot_enabled():
+                        self.port_dict[lport]['cmis_apsel'] = appl
+                        sfp.set_cmis_application_stop(host_lanes)
+                        self.port_dict[lport]['cmis_state'] = self.CMIS_STATE_DP_DEINIT
+                    elif state == self.CMIS_STATE_DP_DEINIT:
+                        if sfp.get_module_state() != 'ModuleReady':
+                            continue
+                        appl = self.port_dict[lport]['cmis_apsel']
+                        sfp.set_cmis_application_apsel(host_lanes, appl)
+                        self.port_dict[lport]['cmis_state'] = self.CMIS_STATE_AP_CONF
+                    elif state == self.CMIS_STATE_AP_CONF:
+                        st = sfp.get_cmis_state()['config_state']
+                        done = True
+                        for lane in host_lanes:
+                            name = "ConfigStatusLane{}".format(lane + 1)
+                            if st[name] != 'ConfigSuccess':
+                                done = False
+                                continue
+                        if not done:
+                            continue
+                        sfp.set_cmis_application_start(host_lanes)
+                        self.port_dict[lport]['cmis_state'] = self.CMIS_STATE_DP_INIT
+                    elif state == self.CMIS_STATE_DP_INIT:
+                        st = sfp.get_cmis_state()['datapath_state']
+                        done = True
+                        for lane in host_lanes:
+                            name = "DP{}State".format(lane + 1)
+                            if st[name] != 'DataPathInitialized':
+                                done = False
+                                continue
+                        if not done:
+                            continue
+                        sfp.set_cmis_application_txon(host_lanes)
+                        self.port_dict[lport]['cmis_state'] = self.CMIS_STATE_DP_TXON
+                    elif state == self.CMIS_STATE_DP_TXON:
+                        st = sfp.get_cmis_state()['datapath_state']
+                        done = True
+                        for lane in host_lanes:
+                            name = "DP{}State".format(lane + 1)
+                            if st[name] != 'DataPathActivated':
+                                done = False
+                                continue
+                        if not done:
+                            continue
+                        state = self.CMIS_STATE_READY
+                        self.dbg_print("{}: {}G, {}-lanes, state={}".format(
+                                       lport, int(speed/1000), len(host_lanes), state))
+                        self.port_dict[lport]['cmis_state'] = state
+
+                except (NotImplementedError, AttributeError):
+                    self.port_dict[lport]['cmis_state'] = self.CMIS_STATE_FAILED
+
+        self.dbg_print("Stopped")
+
+    def task_run(self, y_cable_presence):
```

Comment:
what's the purpose of having "y_cable_presence"  here?

Reply:
Yep, you're right, I'll remove it